### PR TITLE
Remove scaling factor (alpha) in Focal Loss

### LINKED ISSUE #6517: Scaling Term is multiplied twice on calculating Focal Loss
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Both BCELoss[1] and FocalLoss[2] have a weight factor.
In `BCELoss`, we can control it by `pos_weight` parameter, whereas in `FocalLoss`, we can control it by `alpha`.
This means that in the YOLOv5 code base, weight factor is computed twice.
This is a mistake.

Reference:

* [1] BCEWithLogitsLoss: https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html
* [2] FocalLoss: https://arxiv.org/pdf/1708.02002.pdf

### Environment

- YOLO: YOLOv5 (latest; how to check the yolo version?)
- CUDA: 11.6 (Tesla T4, 15360MiB)
- OS: Ubuntu 18.04.6 LTS (Bionic Beaver)
- Python: 3.8.12

### Minimal Reproducible Example

BCE loss and FocalLoss(gamma=0.0) are different value. They should be equal.

```console
$ python test.py                                                                                                                                                   (pytorch_x86)
...
assert FocalLoss_before(BCE(pw=1), gamma=0) != BCE(pw=1)
ok
```
[test.py.zip](https://github.com/ultralytics/yolov5/files/7990662/test.py.zip)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by bilzard**
close #6517

## Note

I conducted unit-test on my local machine: https://github.com/ultralytics/yolov5/pull/6468#issuecomment-1024883777

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced Focal Loss implementation for better training stability.

### 📊 Key Changes
- Removed the `alpha` parameter from the FocalLoss class.
- Adjusted the forward method to exclude the `alpha_factor` in loss calculation.

### 🎯 Purpose & Impact
- This change simplifies the Focal Loss function, potentially improving the stability and performance of models during training.
- The removal of the `alpha` parameter means users no longer need to tune this hyperparameter, possibly making the loss function easier to use and understand.
- The models trained using this revised loss function might yield more consistent results and potentially require less parameter tuning from the user's side. 🎛️

### CODE REVIEW

**Comment by bilzard** [comment]
I find an issue. In the first place, as weight factor is already weighted on the BCELoss, we shouldn't calculate it again in FocalLoss. Let me think how to fix.

**Comment by bilzard** [comment]
> I find an issue. In the first place, as weight factor is controlled by alpha on FocalLoss, the BCELoss should be instantiated by pos_weight=1. Let me think of fixed implementation.

I decided to remove alpha factor because it is already considered on BCELoss.
Please check above discussion and let me know if I am wrong.

**Comment by bilzard** [comment]
FYI: I committed removing alpha. I also fixed description of the PR.

**Comment by bilzard** [comment]
[updated]

I conducted unit-test on my local machine. They are all passed.

output:
```
$ python test.py
assert FocalLoss(BCE(pw=1), gamma=0) == BCE(pw=1)
ok
assert FocalLoss(BCE(pw=2), gamma=0) == BCE(pw=2)
ok
assert FocalLoss(BCE(pw=1), gamma=1) == 2 * FocalLoss_before(BCE(pw=1, gamma=1, alpha=0.5))
ok
assert FocalLoss(BCE(pw=3), gamma=2) == 4 * FocalLoss_before(BCE(pw=1), gamma=2, alpha=0.75)
ok
```

source code:
[test.py.zip](https://github.com/ultralytics/yolov5/files/7963524/test.py.zip)

**Comment by bilzard** [comment]
FYI: I move description to issue #6517.

**Comment by bilzard** [comment]
@glenn-jocher Hi. How is the review going? Do you have any questions?

**Comment by bilzard** [comment]
@glenn-jocher ping

**Comment by glenn-jocher** [comment]
@bilzard hi there! Thanks for reaching out. I've seen your latest updates and appreciate your efforts. We'll ensure to review the changes as soon as possible. Your inputs are valuable to us. Thanks!
